Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectPages.cs

```diff
@@ -61,8 +61,13 @@ public override TokenParser ProvisionObjects(Web web, ProvisioningTemplate templ
                             {
                                 scope.LogDebug(CoreResources.Provisioning_ObjectHandlers_Pages_Overwriting_existing_page__0_, url);
 
-                                if (page.WelcomePage && url.Contains(web.RootFolder.WelcomePage))
+                                string welcomePageUrl = UrlUtility.Combine(web.ServerRelativeUrl, page.ParentTemplate.WebSettings.WelcomePage);
```

Comment:
Hi Bert,

thank you for the feedback - I still have to get a deeper understanding on how everything works so please bear with me ;)

You are correct, I should add a check if the welcome page is defined. 

As I want to consider update scenarios (where the template does not necessarily define WelcomePage although the web has one set) I don't want to rely on the web settings handler to set the WelcomePage back to the correct value.